### PR TITLE
add CMake externalproject support

### PR DESCRIPTION
**Pull request by lishuai87**
we want to build g3log, g3sinks as externalproject.  like this:

```
externalproject_add(g3log
        DOWNLOAD_DIR ${CMAKE_SOURCE_DIR}/tarball
        DOWNLOAD_NAME g3log-1.3.2.tar.gz
        URL https://github.com/KjellKod/g3log/archive/1.3.2.tar.gz
        URL_MD5 9c7d427b4624530c62914f77e87d3d55

        CMAKE_ARGS "-DG3_SHARED_LIB=OFF"
        CMAKE_ARGS "-DUSE_DYNAMIC_LOGGING_LEVELS=ON"
        CMAKE_ARGS "-DCMAKE_INSTALL_PREFIX=${_TMP_INSTALL_PREFIX}"
        )

externalproject_add(g3sinks
        DEPENDS g3log
        DOWNLOAD_DIR ${CMAKE_SOURCE_DIR}/tarball
        DOWNLOAD_NAME g3sinks-1.1.1.tar.gz
        URL https://github.com/KjellKod/g3sinks/archive/1.1.1.tar.gz
        URL_MD5 06a8effd24befdf72b51ab3b89693f21

        CONFIGURE_COMMAND ${CMAKE_COMMAND}
        "-DBUILD_STATIC=ON"
        "-DCMAKE_INSTALL_PREFIX=${_TMP_INSTALL_PREFIX}"
        "-DCMAKE_CXX_FLAGS=-isystem ${_TMP_INSTALL_PREFIX}/include"
        "-DCMAKE_EXE_LINKER_FLAGS=-L ${_TMP_INSTALL_PREFIX}/lib"
        <SOURCE_DIR>/logrotate
        )
```

### CODE REVIEW

**Comment by lishuai87** [comment]
@KjellKod please take a look

**Comment by KjellKod** [comment]
Thanks for the PR. I should be able to get to it, verify and test early next week

File: logrotate/CMakeLists.txt [hunk 1]
```diff
@@ -142,7 +143,7 @@ ENDIF()
 
 SET(${LIBRARY_TO_BUILD}_VERSION_STRING ${VERSION})
 SET_TARGET_PROPERTIES(${LIBRARY_TO_BUILD} PROPERTIES LINKER_LANGUAGE CXX)
-TARGET_LINK_LIBRARIES(${LIBRARY_TO_BUILD}  ${Boost_LIBRARIES} ${ZLIB_LIBRARIES} ${G3LOG})
+TARGET_LINK_LIBRARIES(${LIBRARY_TO_BUILD}  ${Boost_LIBRARIES} ${ZLIB_LIBRARIES} ${G3LOG_LIBRARY})
```
Comment (KjellKod) [inline logrotate/CMakeLists.txt]: Finally got around to review this. 
To make this cohesive can you updated 
```
find_library(g3logger g3logger PATHS /usr/local/probe/lib )
 include_directories(${CMAKE_PREFIX_PATH}/include)
```
 to be 
```
find_library(G3LOG_LIBRARY g3logger PATHS /usr/local/probe/lib )
include_directories(${CMAKE_PREFIX_PATH}/include)
```

And line 14 can be deleted
```
SET(G3LOG_LIBRARY g3logger)
```

Comment (lishuai87) [reply logrotate/CMakeLists.txt]: updated :)

**Comment by KjellKod** [comment]
👍

**Pull request merged**